### PR TITLE
Fixup quickmenu UX

### PR DESCRIPTION
**Pull request by irgolic**
The main reason I wanted to remove categories from the quickmenu by default is because it was easy to hover over a category accidentally while trying to select an item, as the menu typically appears to the bottom-**right** of the cursor.

From what I understand, this is also the reason 'sloppy' category selection logic was implemented, such that a category is only selected after a certain time of hovering over it.

Moving the categories to the right of the menu alleviates this problem.

Screenshot 2020-09-23 at 17 19 14

### CODE REVIEW

**Comment by markotoplak** [comment]
I understand the problem, but having categories on the right would be more intuitive for scripts that are written right-to-left. For me the acceptable category positions are on the left or top (obviously not possible).

How about popping this menu up a bit more to the left of the current cursor? Would that be too confusing?

Or how about removing category change on hover? That would be much more predictable as hover. What we currently do is probably not an established UI pattern. On hover I would only expect additional info, not that a list would actually change.

**Comment by irgolic** [comment]
I've
- offset the popup position
- disabled hover selecting categories while there's a search query

This ought to do the trick. Anyone care to give it a go? It feels oddly natural to not have to vertically aim at menu items, as opposed to diagonally.

**Review by ajdapretnar — APPROVED** [review]

File: orangecanvas/document/quickmenu.py [hunk 1]
```diff
@@ -1602,6 +1555,14 @@ def exec_(self, pos=None, searchText=""):
         triggered. 'Search' text field is initialized with `searchText` if
         provided.
         """
+        # to avoid accidental category hovers, offset the quickmenu
+        # these were calculated by hand, the actual values can't be grabbed
+        # before showing the menu for the first time (and they're defined in qss)
+        x_offset = 33
+        if self.__pages.navigator.categoriesEnabled():
+            x_offset += 33
```
Comment (janezd) [inline orangecanvas/document/quickmenu.py:1563]: This doesn't work well if the popup would stick outside the right margin of the screen - it is shown 33 px left of the mouse.

Comment (irgolic) [reply orangecanvas/document/quickmenu.py:1563]: Oh wow, nice catch.

**Pull request merged**